darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -216,6 +224,349 @@ def test_manual_save_and_load(self):
+            os.mkdir(checkpoint_path_manual)
+            checkpoint_file_name = "checkpoint_0.pth.tar"
+            model_path_manual = os.path.join(
+                checkpoint_path_manual, checkpoint_file_name
+            )
+
+            # define encoders sets
+            encoders_past = {
+                "datetime_attribute": {"past": ["day"]},
+                "transformer": Scaler(),
+            }
+            encoders_other_past = {
+                "datetime_attribute": {"past": ["hour"]},
+                "transformer": Scaler(),
+            }
+            encoders_past_noscaler = {
+                "datetime_attribute": {"past": ["day"]},
+            }
+            encoders_past_other_transformer = {
+                "datetime_attribute": {"past": ["day"]},
+                "transformer": BoxCox(),
+            }
+            encoders_2_past = {
+                "datetime_attribute": {"past": ["hour", "day"]},
+                "transformer": Scaler(),
+            }
+            encoders_past_n_future = {
+                "datetime_attribute": {"past": ["day"], "future": ["dayofweek"]},
+                "transformer": Scaler(),
+            }
+
+            model_auto_save = create_DLinearModel(
+                auto_name, save_checkpoints=True, add_encoders=encoders_past
+            )
+            model_auto_save.fit(self.series, epochs=1)
+
+            model_manual_save = create_DLinearModel(
+                manual_name, save_checkpoints=False, add_encoders=encoders_past
+            )
+            model_manual_save.fit(self.series, epochs=1)
+            model_manual_save.save(model_path_manual)
+
+            model_auto_save_other = create_DLinearModel(
+                auto_name_other, save_checkpoints=True, add_encoders=encoders_other_past
+            )
+            model_auto_save_other.fit(self.series, epochs=1)
+
+            # prediction are different when using different encoders
+            self.assertNotEqual(
+                model_auto_save.predict(n=4),
+                model_auto_save_other.predict(n=4),
+            )
+
+            # model with undeclared encoders
+            model_no_enc = create_DLinearModel("no_encoder", add_encoders=None)
+            # weights were trained with encoders, new model must be instantiated with encoders
+            with self.assertRaises(ValueError):
+                model_no_enc.load_weights_from_checkpoint(
+                    auto_name,
+                    work_dir=self.temp_work_dir,
+                    best=False,
+                    load_encoders=False,
+                )
+            # overwritte undeclared encoders
+            model_no_enc.load_weights_from_checkpoint(
+                auto_name,
+                work_dir=self.temp_work_dir,
+                best=False,
+                load_encoders=True,
+            )
+            # equality between data transformers is not well defined
+            self.assertEqual(
```

Comment:
the comparison of encoders and transformers can be put into a helper function as it occurs 3 times